Integrate AliTPCCommon into O2 + merge new features, 2nd attempt

davidrohr opened the pull request:
Integrates AliTPCCommon (or now called GPU) into O2 (new attempt of AliceO2Group/AliceO2#1747)
- Merges AliTPCCommon into O2
- Renames AliTPCCommon to GPU
- Includes all AliTPCCommon patches between 2.3.2.1 and 2.5 + related O2 patches, which could not be merged before due to problems in AliRoot.
- Contains new ITS GPU track fit jointly developed in both repositories.
- Contains new TPC dEdx Code for GPUs.
- Squashed all commits not related to TPC tracking / space charge / transformation.

ping @ehellbar @matthiasrichter @sawenzel @dberzano @martenole

dberzano commented:
This was opened as agreed. We may have fixed the problem with the C.I.. I would fix the problems currently reported by Travis before proceeding.

davidrohr commented:
@dberzano : I am right now checking the codacy things, then I'll fix the clang-format stuff (apparently the travis clang-format still yields other results as my clang-format, even though it's the same version....)

For the doxygen, I'll have to see how far I get. Not really familiar with it.
I'd suggest you leave me some time, also to test with AliRoot, and then we merge Thursday morning as you suggested.

mconcas commented:
Hi @dberzano @davidrohr, thanks for resuming the work on this.
What about test this PR the new CUDA compilation test?

davidrohr commented:
@mconcas : With this PR, the new CUDA compilation should work. But I'd go step by step, let's get this done first, then we can merge the CUDA autodetection, with some luck also on Thursday.

mconcas commented:
@davidrohr ok, that sounds good to me. Hence, I will try to rebase my work as soon as this is merged. BTW, I'm still developing with CUDA 10.0 (doesn't really care to me the 10.1) but I would like to see my next PR tested with the CUDA builder, so I definitely need this! 
Thanks for the effort again!

davidrohr commented:
@dberzano : This is in principle ready to be merged, just rebased and applied minor fixes.
- clang-format: done
- license: moved to O2 license
- doxygen: I am no expert, but I fixed what seems possible. The main problem seems to be that doxygen does not undestand the macro and template magic needed for OpenCL. I don't see a way to get rid of that. We could only remove all doxygen commands from the former HLT code, that might get rid of the warnings.
- codacy: fixed everything that seems reasonable, the rest either complains about unused / AliRoot files, or are false alarms.
- codechecker: all issues in the GPU dir fixed, remaining errors are from includes from somewhere else.

dberzano commented:
Very good, let me have a look.

dberzano commented:
Fine for me. I am merging, and out of completeness I am restarting the C.I. as well.
